modules/openapi-generator/src/main/resources/go/model_anyof.mustache

```diff
@@ -68,7 +68,7 @@ func (dst *{{classname}}) UnmarshalJSON(data []byte) error {
 }
 
 // Marshal data from the first non-nil pointers in the struct to JSON
-func (src *{{classname}}) MarshalJSON() ([]byte, error) {
+func (src {{classname}}) MarshalJSON() ([]byte, error) {
```

Comment:
thanks for the PR

do we need to apply the same fix model_oneof.mustache (oneOf template) as well?

Reply:
No, it already uses a value receiver. I searched for other occurances using this:

```
$ grep -ie "func (\w\+ \*.*) MarshalJSON" ./modules/openapi-generator/src/main/resources/go/*
```

Which only gives a hit in `model_anyof.mustache`.

Reply:
thanks for the confirmation.

can you please add a test or 2 in `samples/openapi3/client/petstore/go/model_test.go`?

you may want to update `modules/openapi-generator/src/test/resources/3_0/go/petstore-with-fake-endpoints-models-for-testing-with-http-signature.yaml` with new test schemas (e.g. anyOf)